pkg/operator/ceph/disruption/clusterdisruption/pools.go

```diff
@@ -130,7 +130,14 @@ func (r *ReconcileClusterDisruption) getOngoingDrains(request reconcile.Request)
 			nodeList := &corev1.NodeList{}
 			err = r.client.List(context.TODO(), nodeList, client.MatchingLabels{corev1.LabelHostname: nodeHostname})
 			nodeNum := len(nodeList.Items)
-			if err != nil || nodeNum < 1 {
+			if nodeNum < 1 {
+				logger.Infof("node %s does not exist, deleting corresponding drain-canary", nodeHostname)
+				err = r.client.DeleteAllOf(context.TODO(), &appsv1.Deployment{}, client.MatchingLabels{corev1.LabelHostname: nodeHostname, k8sutil.AppAttr: nodedrain.CanaryAppName})
+				if err != nil {
+					logger.Errorf("could not delete drain-canary for host %s: %+v", nodeHostname, err)
```

Comment:
If there is not a return or continue statement after an error, we need a comment to clarify that we expect to continue execution even when there is an error. And maybe it's only a warning instead of an error if we can continue execution?

Reply:
will change to return